src/ShadowDOM/querySelector.js

```diff
@@ -136,6 +136,7 @@
     var target = unsafeUnwrap(this);
     var list;
     var root = getTreeScope(this).root;
+    root = (!deep && root.shadowRoot) || root;
```

Comment:
I think it would be much harder to read, since `root` is used on both sides of `||`, also you'll need to call it anyway regardless of `deep` value.